groovy-language-server/src/main/java/com/palantir/groovylanguageserver/GroovycWrapper.java

```diff
@@ -116,6 +130,48 @@ public Path getWorkspaceRoot() {
     }
 
     @Override
+    public Map<String, Set<SymbolInformation>> getReferences() {
+        return references;
+    }
+
+    @Override
+    public Set<SymbolInformation> findReferences(ReferenceParams params) {
+        Set<SymbolInformation> symbols = fileSymbols.get(params.getTextDocument().getUri());
+        if (symbols == null) {
+            return Sets.newHashSet();
+        }
+        Set<SymbolInformation> foundSymbolInformations = symbols.stream()
+                .filter(s -> {
+                    // If we are given a position that is located within the first line of this symbol's location, then
+                    // this is its definition and is likely to be the symbol we are trying to find.
+                    Position startPosition = s.getLocation().getRange().getStart();
```

Comment:
So that doesn't work because a range is a series of positions, so you can't just check that they're the same, hence why we check contains. Class symbol ranges don't just cover where their name is defined, but their whole block (from keyword class to their closing curly brace).